doc/api/vm.markdown

```diff
@@ -301,6 +301,38 @@ e.g. `(0,eval)('code')`. However, it also has the following additional options:
 - `timeout`: a number of milliseconds to execute `code` before terminating
   execution. If execution is terminated, an [`Error`][] will be thrown.
 
+## Example: Run a Server within a VM
+
+The context of `.runInThisContext()` refers to the v8 context. The code passed
+to this VM context will have it's own isolated scope. To run a simple web server
+using the `http` module, for instance, the code passed to the context must either
+call `require('http')` on it's own, or have a reference to the `http` module passed
+to it. For instance:
+
+```js
+'use strict'
+const vm = require('vm')
+
+let code = `
+  (function(require) {
+
+    const http = require('http');
+
+    http.createServer( (request, response) => {
+      response.writeHead(200, {'Content-Type': 'text/plain'});
+      response.end('Hello World\\n');
+    }).listen(8124);
+
+    console.log('Server running at http://127.0.0.1:8124/');
+  })
+`
+
+vm.runInThisContext(code)(require)
+```   
+
+_Note: `require()` in the above case shares the state with context it is passed
+from. This might introduce risks when unknown code is executed._
```

Comment:
Can we expand this a little bit? Personally, I would love to know the potential problems, if there is risk involved. Unknown code here means untrusted code or user supplied code?